framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -472,6 +485,29 @@ private static int printClassDefinitions(
         return classNames.length;
     }
 
+    /**
+     * Constructs an array of TypeElements corresponding to the list of classes.
+     *
+     * @param innermostTypeElt the innermost type element: either an inner class or an outer class
+     *     without any inner classes that should be printed
+     * @param classNames the names of the containing classes, from outer to inner
+     * @return an array of TypeElements whose entry at a given index represents the type named at
+     *     that index in {@code classNames}
+     */
+    private static TypeElement @SameLen("#2") [] getTypeElementsForClasses(
+            TypeElement innermostTypeElt, String[] classNames) {
+        if (classNames.length == 1) {
```

Comment:
Nit: I wouldn't use the special case.  This isn't speed-critical code, and a special case gives rise to the possibility of error or incompatibilities.  I think the general case works, so this is not necessary for correctness.